container/type.go

```diff
@@ -8,3 +12,13 @@ const (
 	STOPPED StatusType = 1
 	RUNNING StatusType = 2
 )
+
+// TimeoutError represents an error of timeout
```

Comment:
I think this is not a really good way to have errors I like the fact that we have the details in the error message but i'm sure we can do it better, I would just use something like that directly
```go
errors.New(Timeout reached after " + e.duration.String() + " for ressource " + e.name)
```

or maybe have a function
```go
func timeoutError(duration, resource) error {
  return errors.New(Timeout reached after " + e.duration.String() + " for ressource " + e.name)
}
```

or just a constant with generic error message but like that we can compare this error in the tests for example.
```go
const TimeoutError = errors.New("Timeout error on container action")
```

This is a bit too much and confusing because you don't even return an error

Reply:
It's the classic way to create new errors in go and be able to compare. Eg with type casting:
```go
_, ok := err.(*TimeoutError)
```
https://blog.golang.org/error-handling-and-go

On the contrary, I think we should start creating custom error than using the default one with just a custom description. It will be way easier to debug.

Reply:
it just feels like a lot of code just for a simple error that could be done in one constant but ok, let's follow the guidelines